### PR TITLE
docs: add linters description

### LINKED ISSUE #3927: Add description for linters to website
### Your feature request related to a problem? Please describe.

The issue is about enhancing the documentation of each linter's [configuration](https://golangci-lint.run/usage/linters). Currently, the reader has to scroll to the top of the page to understand what each linter does. The goal is to simplify this process by adding a description of each linter directly within its configuration, eliminating the need for the reader to jump to the top of the page.

### Describe the solution you'd like.

Here is how a description can look for ginkgolinter and [gocognit](https://golangci-lint.run/usage/linters/#gocognit):
image

### PR DESCRIPTION
**Pull request by AlduLonghi**
## What?

resolves #3927 

The idea was to add the description for each linter before the snippet in the /linters section.

**Additional**
I added a formatting function that normalizes all linters' descriptions. This function uppercases the first letter and adds a final dot if missing.

## Why?

This is for better understanding so the user doesn't have to scroll all the way up to read the linter's description.

## How?

I call the function GetAllSupportedLinterConfigs() and create a map with each linter's name and description. This way I decrease the complexity since I use just one loop to make it and the later search is O(1).

## Testing?

Everything is working fine.

## Screenshots

From this:
Captura de pantalla 2023-07-09 175357

To this: 
Captura de pantalla 2023-07-09 175328

**Additional**

From this:
Captura de pantalla 2023-07-09 175216

To this:
Captura de pantalla 2023-07-09 175236

Looking forward to your comments :)

### CODE REVIEW

File: scripts/expand_website_templates/main.go [hunk 1]
```diff
@@ -302,7 +302,7 @@ func getDesc(lc *linter.Config) string {
 		}
 	}
 
-	return strings.ReplaceAll(desc, "\n", "<br/>")
+	return capitalizeAndAddFinalDot(strings.ReplaceAll(desc, "\n", "<br/>"))
```
Comment (ldez) [inline scripts/expand_website_templates/main.go:305]:
```suggestion
	return formatDesc(desc)
```

File: scripts/expand_website_templates/main.go [hunk 2]
```diff
@@ -570,3 +578,14 @@ func marshallSnippet(node *yaml.Node) (string, error) {
...
+func capitalizeAndAddFinalDot(dsc string) string {
+	runes := []rune(dsc)
+	runes[0] = unicode.ToUpper([]rune(dsc)[0])
+	dsc = string(runes)
+	if dsc[len(dsc)-1:] != "." {
+		dsc = fmt.Sprintf("%s.", dsc)
+	}
+
+	return dsc
+}
```
Comment (ldez) [inline scripts/expand_website_templates/main.go:591]:
```suggestion
func formatDesc(desc string) string {
	runes := []rune(desc)
	runes[0] = unicode.ToUpper(rune(desc[0]))

	if runes[len(runes)-1] != '.' {
		runes = append(runes, '.')
	}

	return strings.ReplaceAll(string(runes), "\n", "<br/>")
}
```

**Review by ldez — CHANGES_REQUESTED** [review]

File: scripts/expand_website_templates/main.go [hunk 3]
```diff
@@ -16,13 +16,13 @@ import (
...
 	"strings"
-
-	"gopkg.in/yaml.v3"
+	"unicode"
 
 	"github.com/golangci/golangci-lint/internal/renameio"
 	"github.com/golangci/golangci-lint/pkg/config"
 	"github.com/golangci/golangci-lint/pkg/lint/linter"
 	"github.com/golangci/golangci-lint/pkg/lint/lintersdb"
+	"gopkg.in/yaml.v3"
```
[the remark this answers is not in the document]
Comment (AlduLonghi) [reply scripts/expand_website_templates/main.go]: Done

File: scripts/expand_website_templates/main.go (hunk 1 above)
Comment (AlduLonghi) [reply scripts/expand_website_templates/main.go]: Done

File: scripts/expand_website_templates/main.go (hunk 2 above)
Comment (AlduLonghi) [reply scripts/expand_website_templates/main.go]: Done

**Review by ldez — APPROVED** [review]
Hello,

Thank you for your contribution to our project.
Your PR will benefit many users.

I watched your profile and you seem to have frontend knowledge, we have currently a problem with gatsby in the documentation build (#3864, #3934), can you help us with that?

**Comment by AlduLonghi** [comment]
Hey @ldez! Thank you.

I do have frontend knowledge but I actually never worked with Gatsby before. But let me take a look at the issues and I'll see what I can do about them :)

**Comment by ldez** [comment]
@AlduLonghi have you taken a look? Can you help us?

If you cannot it's not a problem, I'm just trying to use the opportunity :smiley_cat:.

**Comment by AlduLonghi** [comment]
Hi @ldez! I've been exploring it the past week and it has to do with Gatsby Migration from v4 to v5. There are some plugins that are not migrated to their latest versions (some needed to be migrated during gatsby migration v3 - v4) and they need code refactoring to function correctly. Is this just this rabbit hole of migrating everything and checking everything works correctly. I might be able to take this next week if that's ok :)

**Pull request merged**